File: Server/Common/NetMessage.php
<?php
	require_once 'OpCode.php';
	class MsgList
	{
		public $msglist = array();
		public $msgnum;
		
		function __construct()
		{
			$this->msgnum = 0;
		}
		
		function addMsg($msg)
		{
			if($msg == NULL) return;
			array_push($this->msglist,$msg);
			$this->msgnum += 1;
		}
		
		function clear()
		{
			if($this->msglist != NULL)
			{
				array_splice($this->msglist,0);
			}
			$this->msgnum = 0;
		}
	}
	
	class NetMessage
	{
		public $opcode;
		
		function __construct($opcode)
		{
			$this->opcode = $opcode;
		}
	}
	
	class COPC_CS_TEST extends NetMessage
	{
		public $testData;
		
		function __construct($opcode = OpCode::OPC_CS_TEST)
		{
			parent::__construct($opcode);
			$this->testData = 110;
		}
	}
	
	class COPC_SC_TEST extends NetMessage
	{
		public $name;
		function __construct($opcode = OpCode::OPC_SC_TEST)
		{
			parent::__construct($opcode);
			$this->name = "water";
		}
	}

File: Client/Common/NetMessage.py
'''
Created on 2013年12月4日

@author: waterkit
'''
from Common.OpCode import *

class MsgList:
    def __init__(self):
        self.msgnum = 0
        self.msglist = []
       
    def addMsg(self,msg):
        if msg == None:
            return
        self.msglist.append(msg.toJsonDict());
        self.msgnum += 1
        
    def clear(self):
        del self.msglist[0:]
        self.msgnum = 0 
        
    def toJsonDict(self):
        jsonDict = {'msgnum':self.msgnum,'msglist':self.msglist}
        #print(jsonDict);
        return jsonDict
    
    def fromJsonDict(self,jsonDict):
        if 'msgnum' not in jsonDict.keys():
            return
        self.msgnum = jsonDict['msgnum']
        
        if 'msglist' not in jsonDict.keys():
            return
        self.msglist = jsonDict['msglist']
        

class NetMessage:
    def __init__(self,opcode):
        self.__opcode = opcode
        self.jsonDict = {}
        
    def setOpcode(self,opcode):
        self.__opcode = opcode;
    
    def getOpcode(self):
        return self.__opcode;
    
    def toJsonDict(self):
        self.jsonDict['opcode'] = self.__opcode
        return self.jsonDict
    
    def fromJsonDict(self,jsonDict):
        if 'opcode' not in jsonDict.keys():
            return
        self.setOpcode(jsonDict['opcode'])
        
class COPC_CS_TEST(NetMessage):
    def __init__(self,opcode = OPC_CS_TEST):
        NetMessage.__init__(self, opcode)
        self.testData = 0
        
    def toJsonDict(self):
        NetMessage.toJsonDict(self)
        self.jsonDict['testData'] = self.testData
        
        return self.jsonDict
    
    def fromJsonDict(self,jsonDict):
        NetMessage.fromJsonDict(self, jsonDict)
        self.testData = self.__jsonDict['testData']
        
class COPC_SC_TEST(NetMessage):
    def __init__(self,opcode = OPC_SC_TEST):
        NetMessage.__init__(self, opcode)
        self.name = ""
        
    def toJsonDict(self):
        NetMessage.toJsonDict(self)
        self.jsonDict['name'] = self.name
        return self.jsonDict
    
    def fromJsonDict(self,jsonDict):
        NetMessage.fromJsonDict(self, jsonDict)
        self.name = jsonDict['name']
    
